crates/holochain_cascade/src/lib.rs

```diff
@@ -905,16 +905,11 @@ impl CascadeImpl {
         })
             .await??;
 
-        let merged_results = results.iter().fold(
-            // It's sort of arbitrary what the initial value is as long as it's
-            // not an activity response.
-            BoundedMustGetAgentActivityResponse::EmptyRange,
-            holochain_types::chain::merge_bounded_agent_activity_responses,
-        );
-
+        let merged_response =
+            holochain_types::chain::merge_bounded_agent_activity_responses(results);
```

Comment:
`merge_bounded_agent_activity_responses` now takes the full list of responses and merges them, rather than merging 2 at a time in a map-reduce.